lib/matplotlib/__init__.py

```diff
@@ -703,35 +696,25 @@ def _get_data_path():
                                'directory')
         return path
 
-    _file = _decode_filesystem_path(__file__)
-    path = os.sep.join([os.path.dirname(_file), 'mpl-data'])
-    if os.path.isdir(path):
-        return path
-
-    # setuptools' namespace_packages may highjack this init file
-    # so need to try something known to be in matplotlib, not basemap
-    import matplotlib.afm
-    _file = _decode_filesystem_path(matplotlib.afm.__file__)
-    path = os.sep.join([os.path.dirname(_file), 'mpl-data'])
-    if os.path.isdir(path):
-        return path
-
-    # py2exe zips pure python, so still need special check
-    if getattr(sys, 'frozen', None):
-        exe_path = os.path.dirname(_decode_filesystem_path(sys.executable))
-        path = os.path.join(exe_path, 'mpl-data')
-        if os.path.isdir(path):
-            return path
-
-        # Try again assuming we need to step up one more directory
-        path = os.path.join(os.path.split(exe_path)[0], 'mpl-data')
-        if os.path.isdir(path):
-            return path
-
-        # Try again assuming sys.path[0] is a dir not a exe
-        path = os.path.join(sys.path[0], 'mpl-data')
-        if os.path.isdir(path):
-            return path
+    # Yield some filenames next to which we may be able to find mpl-data.
+    def get_candidate_siblings():
+        yield __file__
+        # setuptools' namespace_packages may highjack this init file
+        # so need to try something known to be in Matplotlib, not basemap.
+        import matplotlib.afm
+        yield matplotlib.afm.__file__
+        # py2exe zips pure python, so still need special check.
+        if getattr(sys, 'frozen', None):
+            yield sys.executable
+            # Try again assuming we need to step up one more directory.
+            yield os.path.dirname(sys.executable)
```

Comment:
The logic seems wrong for the last two as in the original they did not have `dirname` applied.

Reply:
They do (original: `exe_path = sys.path.dirname(...)`).  For the first one, the candidate will be
```
In [12]: Path(sys.executable).with_name("mpl-data")
Out[12]: PosixPath('/usr/bin/mpl-data')

In [13]: os.path.join(os.path.dirname(sys.executable), "mpl-data")
Out[13]: '/usr/bin/mpl-data'
```